posthog/hogql/database/database.py

```diff
@@ -118,6 +119,9 @@ def create_hogql_database(team_id: int) -> Database:
         database.events.fields["person"] = FieldTraverser(chain=["poe"])
         database.events.fields["person_id"] = StringDatabaseField(name="person_id")
 
+    for mapping in GroupTypeMapping.objects.filter(team=team):
+        database.events.fields[mapping.group_type] = FieldTraverser(chain=[f"group_{mapping.group_type_index}"])
```

Comment:
One change I'd do is to double check if the field name is not already taken. Otherwise users might, accidentally or deliberately, override "system level" fields like "timestamp" or "distinct_id" and cause all manner of problems.

Reply:
Yeep - that's fair. Where do people define the group mappings? Somewhere in the settings?

Reply:
Yep, under project settings:

<img width="1412" alt="image" src="https://github.com/PostHog/posthog/assets/53387/fd9dcaa6-9916-40ad-9504-c7e585cb1c02">